### PR TITLE
SUMO-170423: Create user data source

### PR DESCRIPTION
**Pull request by troy-q-li**
Created data source for user.

### CODE REVIEW

**Comment by troy-q-li** [comment]
Commit too many times for debugging and testing this time. Not the best practice and will avoid it in the future.

File: sumologic/data_source_sumologic_user.go
Comment (sumovishal) [inline sumologic/data_source_sumologic_user.go:56]: `uid` -> `userId` ?

File: sumologic/data_source_sumologic_user.go
Comment (sumovishal) [inline sumologic/data_source_sumologic_user.go:63]: `uemail` -> `userEmail`?

File: sumologic/data_source_sumologic_user.go
Comment (sumovishal) [inline sumologic/data_source_sumologic_user.go:90]: We should call this method `GetUserByEmail`.

File: website/docs/d/user.html.markdown
Comment (sumovishal) [inline website/docs/d/user.html.markdown:11]: "managed by another terraform stack" -> "managed outside of terraform"

File: sumologic/data_source_sumologic_user.go
Comment (sumovishal) [inline sumologic/data_source_sumologic_user.go:73]: "id or name" -> "id or email"

**Review by sumovishal — APPROVED** [review]
lgtm, some minor changes. Approving to unblock merging. Please address comments before merging.

File: website/docs/d/user.html.markdown
Comment (emichaeli) [reply website/docs/d/user.html.markdown:11]: +1
Also, looks like the docs for `role` and `collector` data sources also say `"managed by another terraform stack"`. Maybe update those as well while you're at it? cc @troy-q-li

**Review by emichaeli — APPROVED** [review]
LGTM, nice 👍

**Comment by troy-q-li** [comment]
Address all the comments. Thanks for the review!

**Pull request merged**
